pkg/queue/queue.go

```diff
@@ -80,12 +80,33 @@ func NewQueue[T comparable](opts ...OptionFunc[T]) *Queue[T] {
 // Pull will pull an item from the queue. If the queue is empty, the default value of T will be returned.
 // Note that his does not wait for items to be available in the queue, use PullBlocking instead.
 func (q *Queue[T]) Pull() T {
+	item, _ := q.pull()
+	return item
+}
+
+// PullBlocking will pull an item from the queue, potentially waiting until one is available.
+// In case the waitable signals done, the default value of T will be returned.
```

Comment:
> In case the waitable signals done, the default value of T will be returned

Is this always true, or does that only happen when the queue is empty?

Reply:
My understanding is that this is always true, otherwise we wouldn't be waiting in the select.